nanoFramework.System.Net.Http/Http/System.Net.HttpWebRequest.cs

```diff
@@ -1499,50 +1530,66 @@ private void SubmitRequest()
+                        {
+                            // For HTTPs we still need to know the target name to decide on persistent connection.
+                            m_requestStream = EstablishConnection(proxyUri, m_originalUrl);
+                        }
+                        else
+                        {
+                            // For normal HTTP all requests go to proxy
+                            m_requestStream = EstablishConnection(proxyUri, proxyUri);
+                        }
                     }
                 }
-            }
 
-            // We have connected stream. Set the timeout from HttpWebRequest
-            m_requestStream.WriteTimeout = m_readWriteTimeout;
-            m_requestStream.ReadTimeout = m_readWriteTimeout;
+                // skigrinder - 2020-12-10 make sure m_requestStream is valid
+                if (m_requestStream == null)
+                {
+                    // Connection could not be established
+                    m_requestSent = false;
+                    return;
+                }
+
+                // We have connected stream. Set the timeout from HttpWebRequest
+                m_requestStream.WriteTimeout = m_readWriteTimeout;
+                m_requestStream.ReadTimeout = m_readWriteTimeout;
 
-            // Now we need to write headers. First we update headers.
-            PrepareHeaders();
+                // Now we need to write headers. First we update headers.
+                PrepareHeaders();
 
-            // Now send request string and headers.
-            byte[] dataToSend = GetHTTPRequestData();
+                // Now send request string and headers.
+                byte[] dataToSend = GetHTTPRequestData();
 
 #if DEBUG   // In debug mode print the request. It helps a lot to troubleshoot the issues.
-            int byteUsed, charUsed;
-            bool completed = false;
-            char[] charBuf = new char[dataToSend.Length];
-            UTF8decoder.Convert(dataToSend, 0, dataToSend.Length, charBuf, 0, charBuf.Length, true, out byteUsed, out charUsed, out completed);
-            string strSend = new string(charBuf);
-            Console.WriteLine(strSend);
+                int byteUsed, charUsed;
+                bool completed = false;
+                char[] charBuf = new char[dataToSend.Length];
+                UTF8decoder.Convert(dataToSend, 0, dataToSend.Length, charBuf, 0, charBuf.Length, true, out byteUsed, out charUsed, out completed);
+                string strSend = new string(charBuf);
+                Console.WriteLine(strSend);
 #endif
-            // Writes this data to the network stream.
-            m_requestStream.Write(dataToSend, 0, dataToSend.Length);
-            m_requestSent = true;
+                // Writes this data to the network stream.
+                m_requestStream.Write(dataToSend, 0, dataToSend.Length);
+                m_requestSent = true;
+            }
+            catch
+            {
```

Comment:
Need a comment here explaining why it's OK to have a catch all.

Reply:
done